root/programs/C#/Frameworks/Infrastructure/Framework/Transmission/CallController.cs

```diff
@@ -1253,7 +847,233 @@ public object Invoke(string serviceName, object parameterValue)
 
         #endregion
 
-        // インナークラス（プロキシ類）
+        #region ヘルパー
+
+        /// <summary>CreateProxy</summary>
+        /// <param name="props">Dictionary(string, string)</param>
+        /// <returns>WebProxy</returns>
+        private WebProxy CreateProxy(Dictionary<string, string> props)
+        {
+            WebProxy proxy = null;
+
+            #region プロキシ生成
+
+            if (this._proxyUrl == null || this._proxyUrl == "")
+            {
+                // ユーザ指定：なし
+
+                if (!props.ContainsKey(FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_URL))// Dic化でnullチェック変更
+                {
+                    // XML定義：キーが無い
+                }
+                else
+                {
+                    if (props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_URL] == null
+                        || (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_URL] == "")
```

Comment:
@daisukenishino 

Since the type of variable *props* is `Dictionary<string, string>`, it is unnecessary to cast to `string`.

Furthermore, since the type of `props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_URL]` is `string`, this code can be simplified as follows:

```cs
if (string.IsNullOrEmpty(props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_URL]))
```